### ISSUE TITLE
Plex Cloud: missing artwork

### ISSUE THREAD

**Issue by croneter** [croneter/PlexKodiConnect#189]
Currently, artwork is missing if a Plex Cloud PMS is used:

![screen shot 2017-01-21 at 4 53 09 pm](https://cloud.githubusercontent.com/assets/4637940/22183702/fa3ff0de-e0c3-11e6-83d5-183c77308501.png)
![screen shot 2017-01-21 at 4 51 56 pm](https://cloud.githubusercontent.com/assets/4637940/22183701/fa3f816c-e0c3-11e6-9358-833f778ee153.png)

**Comment by mvanbaak** [comment]
Just wanted to add a 'I am having the same problem.'

Library syncs, but no artwork.

**Comment by croneter** [comment]
Found the issue. 

PKC needs to use links like `https://<PMS>.sjc.plex.services:443/library/metadata/93/thumb/1485817392?X-Plex-Token=<TOKEN>`. Plex Cloud redirects this URL to something like
```
https://<PMS>.sjc.plex.services/services/clouduploader/get?remotePath=Albums/b/<XYZ>.bundle/Contents/_combined/posters/com.plexapp.agents.plexmusic_c826aad0791921ab593b5e433204d42bc6406245&type=1
```
Plex Cloud **drops the url query string arguments like `X-Plex-Token`**. Unfortunately, it's impossible for PKC to use this info in headers, but needs to store everything in the url query string. 

I don't have access to the Plex Cloud beta forum. Could you please be so kind and open an issue for this on the Plex side? Thanks

**Comment by croneter** [comment]
Meaning @mvanbaak - could you please pass this along to the Plex team? :-)

**Comment by mvanbaak** [comment]
@croneter I will. I'll link this issue there.

**Comment by croneter** [comment]
@mvanbaak - is this still an issue?

**Comment by mvanbaak** [comment]
yes. See attached log

**Comment by mvanbaak** [comment]
[missingartwork.log.txt](https://github.com/croneter/PlexKodiConnect/files/834329/missingartwork.log.txt)

**Comment by croneter** [comment]
Any feedback from Plex? Which forum thread was that? :-)

**Comment by mvanbaak** [comment]
http://forums.plex.tv/discussion/256438/plexkodiconnect-is-unable-to-download-artwork-from-plexcloud#latest

**Comment by mvanbaak** [comment]
By the way, I'm still sharing my plex cloud with your test account ;-)

**Comment by croneter** [comment]
Thanks! Silly me ;-)

**Comment by mvanbaak** [comment]
No worries. Bought some extra coffee's for you ;-)

**Comment by croneter** [comment]
Many thanks!! :-D

**Comment by MarhyCZ** [comment]
If the PKC can use Fanart.tv to download additional artwork, what about to use it as a temporary workaround to download all of the artwork when using Plex Cloud?

**Comment by MarhyCZ** [comment]
I poked the PKC thread on Plex forums (http://forums.plex.tv/discussion/256438/plexkodiconnect-is-unable-to-download-artwork-from-plexcloud#latest)

And Plex Employee answered quickly with probably a working solution!

"I've informed the devs. Due to the way Plex Cloud works, that API doesn't quite work properly. With all the other things going on for Plex Cloud, fixing that API is a bit low on the priority list. Alternatively and the proper way to request artwork is to invoke the transcode API. If you set the size high enough, it won't bother converting the image and you will get the full size image, so you basically get the same result.
https://*.sjc.plex.services/photo/:/transcode?width=4000&amp;height=4000&amp;minSize=1&amp;upscale=0&amp;url=/library/metadata/6/thumb/1495650195?X-Plex-Token=xxx&amp;X-Plex-Token=xxx
Will get you the same as if you used
https://*.sjc.plex.services/library/metadata/6/thumb/1495650195?X-Plex-Token=xxx
Assuming the image is smaller than 4,000 x 4,000, which for Plex Cloud, they will be."
